spec.md

```diff
@@ -0,0 +1,238 @@
+  Title is a collection of localized strings, indexed with [ISO Language Codes][iso-lang-codes].
+  - **`locale`** *string*  
+    2-2 language codes representing the country and locale. [ISO Language Codes][iso-lang-codes]
+  - **`title`** *string*  
+Localized title. The max length MUST not exceed 30 characters and MUST not encode formatting characters.
+- **`description`** *string-string map*  
+    This REQUIRED property must have at least one value, providing a short description of the type for human consumption. The description may be displayed in repository listings or registry tooling.  
+    Description is a collection of localized strings, indexed with [ISO Language Codes][iso-lang-codes].  
+  - **`locale`** *string*  
+    2-2 language codes representing the country and locale. [ISO Language Codes][iso-lang-codes]
+  - **`description`** *string*  
+Localized description. The max length MUST not exceed 255 characters and MUST not encode formatting characters.
+- **`moreInfoUrl`** *url*  
+This OPTIONAL property provides additional details, intended for consumers of the type. This is most often a marketing & awareness overview page.
+- **`specUrl`** *url*  
+This OPTIONAL property references a spec, providing additional details on the type.
+- **`tools`** *string-string map*  
+  This OPTIONAL property provides a collection of tools that may be used with artifact type. The property is intended for end users to find more info on how to find and install related tools. Registry operators MAY provide links to the tools in their repository listing.
+  - **`url`** *url*  
+    This REQUIRED property links to a page where users can download the tool. The URL MAY be a direct link to a download URL, or a link to documentation for how to download the tool.
+  - **`title`** *string-string map*  
+    This OPTIONAL property representes the name of the tool, displayed for human consumption. The title may be displayed in a repository listing, or registry tooling.  
+    Title is a collection of localized strings, indexed with [ISO Language Codes][iso-lang-codes].
+    - **`locale`** *string*  
+      2-2 language codes representing the country and locale. [ISO Language Codes][iso-lang-codes]
+    - **`title`** *string*  
+Localized title. The max length MUST not exceed 30 characters and MUST not encode formatting characters.
+- **`configSchemaReferenceUrl`** *url*  
+This OPTIONAL property provides a schema reference for the artifact config object. The schema is provided for registry operators and tools to optionally validate and process information within the config. A registry operator MAY wish to present information, such as the OCI image architecture type. Each versioned artifact type would have a unique version, possibly referencing a unique schema version. To version the schema, the artifactType MUST also be versioned.
+- **`layerMediaTypes`** string-string map  
```

Comment:
This and `configSchemaReferenceUrl` are tightly ties to the manifest media type, which is otherwise fairly generic.  Especially with this field REQUIRED, that will make it difficult to pivot to index and other types in the future.  Since the artifact store has to know about walking descriptor references to create Merkle trees to avoid garbage collecting a dependency blob, maybe just `schema` with a URI for the media type being registered?  Then the registry looks up your manifest media type, and validates it against your manifest spec (which could restrict layer media types).  When it walks down to your config, it looks up the registration for your config type, grabs the referenced JSON Schema, and validates your config payload against that.  That would be fairly generic, with the quibble that you couldn't overload `application/vnd.oci.image.manifest.v1+json` with multiple schemas (which is arguably a good thing).

Reply:
+1. I think having a required schema is necessary to ensure good GC practices, as well as to ensure we don't end up with invalid manifests being pushed into the repositories.

Reply:
I'm trying to understand this comment a little better. 
This section [Optional: Defining Config Schema](#optional-defining-config-schema), is for registering well-known types. Artifact authors MAY provide a `artifactManifest.json` that helps registry operators understand how to represent the type. Artifact authors MAY optionally provide a config object - meaning a value for the config, not just use the mediaType name. If they do define a config object/value, they MUST provide a schema so registry operators MAY parse it. 
Is the feedback that the schema should be included directly within the `artifactManifest.json` because finding the schema is difficult at runtime? 
The assumption we've been making is registry operators have an onboarding process, and they don't reach out realtime to get these values. They would pull in the logos, strings and possibly the schemas, if they wish to parse them.
I'm just trying to understand if the comment was in the scope of the optional `artifactManifest.json` file.

Reply:
kk... hmm let me chat with steve a bit and see if index will be an issue thx @wking

Reply:
I think there's some confusion over the term manifest here. I've used it generically, not suggesting it has anything to do with the `image-manifest` or `image-index`. This is a document that allows an artifact author to publish information about the type. I'll provide more context.

Reply:
I've moved this content to another document. I'll keep the spec.md focused on implementing an artifact type. I'll make a separate PR for `publishing-well-known-types.md`